hathor/manager.py

```diff
@@ -613,37 +624,18 @@ def validate_new_tx(self, tx: BaseTransaction, skip_block_weight_verification: b
             assert tx.hash is not None  # XXX: it appears that after casting this assert "casting" is lost
 
             if not skip_block_weight_verification:
-                # Validate minimum block difficulty
-                block_weight = daa.calculate_block_difficulty(tx)
-                if tx.weight < block_weight - settings.WEIGHT_TOL:
-                    raise InvalidNewTransaction(
-                        'Invalid new block {}: weight ({}) is smaller than the minimum weight ({})'.format(
-                            tx.hash.hex(), tx.weight, block_weight
-                        )
-                    )
-
-            parent_block = tx.get_block_parent()
-            tokens_issued_per_block = daa.get_tokens_issued_per_block(parent_block.get_metadata().height + 1)
-            if tx.sum_outputs != tokens_issued_per_block:
-                raise InvalidNewTransaction(
-                    'Invalid number of issued tokens tag=invalid_issued_tokens'
-                    ' tx.hash={tx.hash_hex} issued={tx.sum_outputs} allowed={allowed}'.format(
-                        tx=tx,
-                        allowed=tokens_issued_per_block,
-                    )
-                )
+                tx.verify_weight()
+
+            tx.verify_reward()
         else:
+            tx = cast(Transaction, tx)
             assert tx.hash is not None  # XXX: it appears that after casting this assert "casting" is lost
-            assert isinstance(tx, Transaction)
```

Comment:
Do the `cast(...)` check whether `isinstance(tx, Transaction)`? I guess it doesn't and we should keep this assert.

Reply:
No, `cast` bypasses any type checks.

Reply:
Done.